app/scenes/channel_drawer/channel_drawer.js

```diff
@@ -27,6 +27,19 @@ export default class ChannelDrawer extends PureComponent {
         currentChannel: {}
     };
 
+    canPress = true;
+
+    handlePress = (action, ...args) => {
+        if (this.canPress) {
+            this.canPress = false;
+            action(...args);
```

Comment:
I could pass as this argument but is really not needed here, and yes maybe not a component but a function that I could use everywhere makes more sense, I'll work something out tomorrow or Monday to do it properly